hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/helpers/TestContainerUtils.java

```diff
@@ -122,6 +127,23 @@ public void testDatanodeIDPersistent(@TempDir File tempDir) throws Exception {
     assertWriteRead(tempDir, id1);
   }
 
+  @Test
+  public void testVerifyInconsistentChecksumShouldFail() throws Exception {
+    ContainerData containerData = mock(ContainerData.class);
+    when(containerData.getContainerType()).thenReturn(
+        ContainerProtos.ContainerType.KeyValueContainer);
+    // mock to return different checksum
+    when(containerData.getChecksum()).thenReturn("checksum1", "checksum2");
+    conf.setBoolean(HddsConfigKeys.HDDS_CONTAINER_CHECKSUM_VERIFICATION_ENABLED, true);
+    try {
+      ContainerUtils.verifyChecksum(containerData, conf);
+      fail("code should not reach here, exception should have been thrown.");
+    } catch (Exception e) {
+      assertTrue(e.getMessage().contains("Container checksum error"));
```

Comment:
use assertThrows to verify exception: https://www.digitalocean.com/community/tutorials/junit-assert-exception-expected

Reply:
yeah agree!